File: benchmarks/tsb/bench_isin.ts
/**
 * Benchmark: Series.isin() — membership test.
 * Outputs JSON: {"function": "isin", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => i % 5000) });
const testSet = Array.from({ length: 2500 }, (_, i) => i);

for (let i = 0; i < WARMUP; i++) {
  s.isin(testSet);
}

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const start = performance.now();
  s.isin(testSet);
  times.push(performance.now() - start);
}

const totalMs = times.reduce((a, b) => a + b, 0);
const meanMs = totalMs / ITERATIONS;
console.log(
  JSON.stringify({
    function: "isin",
    mean_ms: Math.round(meanMs * 1000) / 1000,
    iterations: ITERATIONS,
    total_ms: Math.round(totalMs * 1000) / 1000,
  }),
);

File: benchmarks/pandas/bench_isin.py
"""Benchmark: Series.isin() — membership test."""
import json, time
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

s = pd.Series([i % 5000 for i in range(SIZE)])
test_set = list(range(0, 2500))

for _ in range(WARMUP):
    s.isin(test_set)

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    s.isin(test_set)
    times.append((time.perf_counter() - t0) * 1000)

total_ms = sum(times)
print(json.dumps({"function":"isin","mean_ms":round(total_ms/ITERATIONS,3),"iterations":ITERATIONS,"total_ms":round(total_ms,3)}))
